### PR TITLE
Adds Label and Description components to Typography package

### PR DESCRIPTION
**Pull request by brookescarlett**
## ✍️ Proposed changes

🎟 _Jira ticket:_ [Name of ticket](https://jira.mongodb.org/browse/[name-of-ticket])

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

### CODE REVIEW

File: packages/typography/src/FormElements.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+const labelStyle = css`
+  font-size: 14px;
+  font-weight: bold;
+  line-height: 16px;
+  padding-bottom: 4px;
+`;
+
+type LabelProps = JSX.IntrinsicElements['label'] & {
+  darkMode?: boolean;
+  id: string | number;
```
Comment (lazytype) [inline packages/typography/src/FormElements.tsx:45]: I would call this `inputId` so it doesn't collide with the `id` of the label element itself. should we omit the `htmlFor` property if we're going to rename it? is there a good case for renaming it in the first place?

File: packages/typography/src/FormElements.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+};
+
+const Label = ({ id, darkMode = false, children }: LabelProps) => {
+  const mode = darkMode ? Mode.Dark : Mode.Light;
+
+  return (
+    <label htmlFor={id} className={cx(labelStyle, colorSets[mode].labelColor)}>
```
Comment (lazytype) [inline packages/typography/src/FormElements.tsx:52]: we're not forwarding `className` or the rest of the label props

File: packages/typography/src/FormElements.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+type DescriptionProps = JSX.IntrinsicElements['p'] & {
+  darkMode?: boolean;
+};
+
+const Description = ({ darkMode = false, children }: DescriptionProps) => {
+  const mode = darkMode ? Mode.Dark : Mode.Light;
+
+  return (
+    <p className={cx(descriptionStyle, colorSets[mode].descriptionColor)}>
```
Comment (lazytype) [inline packages/typography/src/FormElements.tsx:77]: we're not forwarding props here either

**Review by lazytype — CHANGES_REQUESTED** [review]

File: packages/typography/src/FormElements.tsx (hunk 1 above)
Comment (brookescarlett) [reply packages/typography/src/FormElements.tsx:45]: Nope, good call... will omit id and just spread htmlFor in rest. Just want to make sure a value for the prop is supplied, so will be sure to type appropriately

File: packages/typography/src/FormElements.tsx (hunk 2 above)
Comment (brookescarlett) [reply packages/typography/src/FormElements.tsx:52]: great call, thank you

File: packages/typography/src/FormElements.tsx (hunk 3 above)
Comment (brookescarlett) [reply packages/typography/src/FormElements.tsx]: fixed

File: packages/typography/src/FormElements.tsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+Label.displayName = 'Label';
+
+const descriptionStyle = css`
+  font-size: 14px;
+  line-height: 16px;
+  font-weight: normal;
+  padding-bottom: 4px;
+  margin-top: 0px;
+  margin-bottom: 0px;
```
Comment (lazytype) [inline packages/typography/src/FormElements.tsx:74]:
```suggestion
  margin-top: 0;
  margin-bottom: 0;
```

**Review by lazytype — APPROVED** [review]

**Pull request merged**